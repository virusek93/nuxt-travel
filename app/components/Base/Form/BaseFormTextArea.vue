<script lang="ts" setup>
const props = defineProps<{
  name: string
  label: string
  placeholder?: string
  customClass?: string
}>()
const { value, errorMessage } = useField<string>(() => props.name)
</script>

<template>
  <BaseFormField
    :name="name"
    :label="label"
    :error-message="errorMessage"
  >
    <textarea
      :id="name"
      v-model="value"
      :name="name"
      :placeholder="placeholder"
      class="min-w-52 rounded-md border border-gray-300 px-4 py-2"
      :class="customClass"
    />
  </BaseFormField>
</template>
