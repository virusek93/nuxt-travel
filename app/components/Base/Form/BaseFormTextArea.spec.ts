import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import BaseFormTextArea from '~/components/Base/Form/BaseFormTextArea.vue'

const props: InstanceType<typeof BaseFormTextArea>['$props'] = {
  name: 'areaName',
  label: 'areaLabel',
  placeholder: 'areaPlaceholder',
  customClass: 'test-custom-class',
}

describe('BaseFormTextArea', () => {
  it('Should render and match the snapshot', () => {
    const wrapper = mount(BaseFormTextArea, { props })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Should pass props to textarea attr', () => {
    const wrapper = mount(BaseFormTextArea, { props })
    const textArea = wrapper.find('textarea')
    expect(textArea.attributes('name')).toBe(props.name)
    expect(textArea.attributes('id')).toBe(props.name)
    expect(textArea.attributes('placeholder')).toBe(props.placeholder)
    expect(
      textArea.attributes('class')?.includes(props.customClass ?? ''),
    ).toBe(true)
  })
})
