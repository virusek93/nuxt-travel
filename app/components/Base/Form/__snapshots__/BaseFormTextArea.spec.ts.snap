// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseFormTextArea > Should render and match the snapshot 1`] = `
<div>
  <label
    class="flex cursor-pointer flex-col items-start gap-0.5 text-sm font-medium text-gray-500"
    data-test="form-label"
    for="areaName"
  >
    <span
      class="after:ml-0.5 after:hidden after:text-red-500 after:content-['*']"
    >
      areaLabel
    </span>
  </label>
  
  <textarea
    class="min-w-52 rounded-md border border-gray-300 px-4 py-2 test-custom-class"
    id="areaName"
    name="areaName"
    placeholder="areaPlaceholder"
  />
  
  <!--v-if-->
</div>
`;
